lib/ocpp/v201/database_handler.cpp

```diff
@@ -51,6 +51,7 @@ void DatabaseHandler::init_enum_table_inner(const std::string& table_name, const
         throw std::runtime_error("Table does not exist.");
     }
 
+    std::scoped_lock lock(this->database_transaction_mutex);
```

Comment:
I would suggest to do this more generically by putting this in the database connection somehow. Apparently you can't have more than 1 transaction per connection so that seems to be the perfect place. Lock on begin transaction and unlock on the commit or rollback.

Reply:
For the record, we discussed this morning I will pick this one up ;)